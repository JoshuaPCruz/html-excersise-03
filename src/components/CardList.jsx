import React from 'react';
import '../css/components/CardList.css';
import Card from './Card';

const CardList = ({cards}) => {
  return (
    <div className="CardList">
      {cards.map((card, index) => (
        <Card image={card.image} title={card.title} body={card.body} key={index} />
      ))}
    </div>
  );
};

export default CardList;