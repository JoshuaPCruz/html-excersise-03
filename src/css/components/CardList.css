.CardList {
  display: flex;
  justify-content: space-around;
  margin-bottom: 50px;
}

.CardList .Card:nth-child(1) h4 {
  color: #496B97;
}

.CardList .Card:nth-child(2) h4 {
  color: #85EBDE;
}

.CardList .Card:nth-child(3) h4 {
  color: #F8C767;
}

.CardList .Card:nth-child(4) h4 {
  color: var(--orange-secondary);
}