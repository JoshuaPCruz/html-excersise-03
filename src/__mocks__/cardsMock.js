import { cards } from '../manifests/CardList.json';

export default cards;