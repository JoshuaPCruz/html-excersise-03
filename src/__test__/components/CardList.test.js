import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import CardList from '../../components/CardList';
import cardsMock from '../../__mocks__/cardsMock';

describe('<CardList />', () => {
  const cardList = shallow(<CardList cards={cardsMock} />);
  test('Render del componente CardList', () => {
    expect(cardList.length).toEqual(1);
  });

  test('Render del <Card /> n times', () => {
    expect(cardList.children()).toHaveLength(cardsMock.length);
  });
});

describe('CardList Snapshot', () => {
  test('Comprobar la UI del componente CardList', () => {
    const cardList = create(<CardList cards={cardsMock} />);
    expect(cardList.toJSON()).toMatchSnapshot();
  });
});