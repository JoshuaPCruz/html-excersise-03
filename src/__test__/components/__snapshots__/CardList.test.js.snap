// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Snapshot Comprobar la UI del componente CardList 1`] = `
<div
  className="CardList"
>
  <figure
    className="Card"
  >
    <img
      alt=""
      src="../src/images/dudes 1.jpg"
    />
    <figcaption>
      <h4>
        Consultoría
      </h4>
      <p>
        Duis autem velm iriure dolor in hendrerit vulputate...
      </p>
    </figcaption>
  </figure>
  <figure
    className="Card"
  >
    <img
      alt=""
      src="../src/images/plant.png"
    />
    <figcaption>
      <h4>
        Formación
      </h4>
      <p>
        Veliesse molestie consequat vel illum dolore...
      </p>
    </figcaption>
  </figure>
  <figure
    className="Card"
  >
    <img
      alt=""
      src="../src/images/images3.png"
    />
    <figcaption>
      <h4>
        Análisis
      </h4>
      <p>
        Feugiat nulla facilisis at vero eroes et accumsan...
      </p>
    </figcaption>
  </figure>
  <figure
    className="Card"
  >
    <img
      alt=""
      src="../src/images/images4.png"
    />
    <figcaption>
      <h4>
        Investigación
      </h4>
      <p>
        Dignissim qui blandit praent luptatum...
      </p>
    </figcaption>
  </figure>
</div>
`;
